.github/workflows/ci.yml

```diff
@@ -24,112 +26,186 @@ jobs:
   build-dev:
     runs-on: mcf-dev-large-x64
     container: mobilecoin/fat-sgx-builder:v24.4.2
-
     steps:
-      - name: Check out code
-        uses: mobilecoinofficial/gh-actions/checkout@v0
-      - name: Cargo build (SW)
-        shell: bash
-        run: cargo build --locked
-      - name: Check dirty git
-        uses: ./.github/actions/check-dirty-git
```

Comment:
was it intentional that these went away?
I'm a bit torn on their value to begin with
I like the idea of ensuring we haven't modified any code, but when we have generated code files that are used in part of the build it doesn't seem as beneficial